contrib/pylightning/lightning/lightning.py

```diff
@@ -714,6 +714,19 @@ def ping(self, peer_id, length=128, pongbytes=128):
         }
         return self.call("ping", payload)
 
+    def plugin(self, subcommand=None, parameter=None):
+        """
+        Manage plugins.
+        """
+        payload = {
+            "subcommand": subcommand
+        }
+        if subcommand in {"stop", "start"}:
+            payload["plugin"] = parameter
+        elif subcommand == "startdir":
+            payload["directory"] = parameter
+        return self.call("plugin", payload)
```

Comment:
Hmm, discarding parameters if we don't know the subcommand seems wrong?  I think we need to whitelist the known parameters?

Reply:
I don't understand, do you mean whitelist on the C side ? Like accepting parameters for `reload` (soon `rescandir` :) and `list` ?
https://github.com/darosior/lightning/blob/09a2dea65292e6713aadeb60e253da8d0bd4f92c/lightningd/plugin_control.c#L81-L95

Reply:
Maybe we should rewrite this into multiple calls that map to the various `plugin` subcommands?

 - plugin_start
 - plugin_stop
 - plugin_rescan
 - etc

That way we have something to attach the specific help messages to instead of handling opaque sets of parameters?